src/Hazel.hs

```diff
@@ -300,13 +301,10 @@ infer dirs ctx t = case t of
 
     case cTmTy of
       WithTy tys -> do
-        assert (idx < V.length tys) (ChooseArg:dirs)
-          "[infer Choose] index mismatch"
-        let usage = snd (head ctx)
-        -- TODO(joel) better api for variable use: strawman: pass in the index
-        -- of the var you want to access, get back the new context.
-        usage' <- useVar (ChooseArg:dirs) usage
-        return (leftovers & _head . _2 .~ usage', tys V.! idx)
```

Comment:
`Choose` doesn't introduce any variable bindings -- it's kind of like array access here. I think we shouldn't have to call `useVar` or update usages.